### PR TITLE
Remove reflection

### LINKED ISSUE #772: Remove Reflection in favor of Member publicize
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description
Before Publicize was used all members were set via reflection (e.g. Extensions folder)

## Intended Design
Find all patches and change to use Publicizer members rather then reflection.

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

### PR DESCRIPTION
**Pull request by MaxDorob**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

Resolved #772

### CODE REVIEW

File: source/Coop.IntegrationTests/Environment/Mock/MockObjectManager.cs [hunk 1]
```diff
@@ -36,7 +36,13 @@ public bool TryGetId(object obj, out string id)
             return false;
         }
 
-        public bool TryGetObject<T>(string id, out T obj) where T : class
+        public bool TryGetNonMBObject<T>(string id, out T obj) where T : class
```
Comment (garrettluskey) [inline source/Coop.IntegrationTests/Environment/Mock/MockObjectManager.cs]: `TryGetNonMBObject` should not exist. Please revert

File: source/Common/Extensions/ReflectionExtensions.cs [hunk 2]
```diff
@@ -183,6 +183,15 @@ public static bool TryCreateDelegate(this Type type, object obj, MethodInfo meth
             { typeof(bool), () => random.Next(2) > 0 ? true : false },
         };
 
+        public static T Random<T>()
```
Comment (garrettluskey) [inline source/Common/Extensions/ReflectionExtensions.cs:191]: Please add summary comments

File: source/GameInterface.Tests/Serialization/SerializerTests/ClanSerializationTest.cs [hunk 3]
```diff
@@ -68,10 +68,10 @@ public void Clan_Full_Serialization()
...
             };
 
-            ClanBinaryPackage.Clan_supporterNotablesCache.SetValue(testClan, heroes);
-            ClanBinaryPackage.Clan_companionsCache.SetValue(testClan, heroes);
-            ClanBinaryPackage.Clan_lordsCache.SetValue(testClan, heroes);
-            ClanBinaryPackage.Clan_supporterNotablesCache.SetValue(testClan, heroes);
+            testClan._supporterNotablesCache = heroes;
+            testClan._companionsCache = heroes;
+            testClan._lordsCache = heroes;
+            testClan._supporterNotablesCache = heroes; // Again?
```
Comment (garrettluskey) [inline source/GameInterface.Tests/Serialization/SerializerTests/ClanSerializationTest.cs]: Not needed twice

```suggestion
```

File: source/GameInterface.Tests/Stubs/ObjectManagerAdapterStub.cs [hunk 4]
```diff
@@ -71,18 +71,25 @@ public bool TryGetId(object obj, out string id)
             throw new NotImplementedException();
         }
 
-        public bool TryGetObject(string id, out object obj) => registry.TryGetValue(id, out obj);
-
-        public bool TryGetObject<T>(string id, out T obj) where T : class
+        public bool TryGetNonMBObject<T>(string id, out T obj) where T : class
```
Comment (garrettluskey) [inline source/GameInterface.Tests/Stubs/ObjectManagerAdapterStub.cs]: Revert these changes please

File: source/GameInterface/Services/Clans/Patches/ClanNameChangePatch.cs [hunk 5]
```diff
@@ -36,8 +32,7 @@ static bool Prefix(ref Clan __instance, TextObject name, TextObject informalName
         public static void RunOriginalChangeClanName(Clan clan, TextObject name, TextObject informalName)
         {
             if (clan == null) return;
-            Clan_Name_Setter(clan, name);
-            Clan_InformalName_Setter(clan, informalName);
```
Comment (garrettluskey) [inline source/GameInterface/Services/Clans/Patches/ClanNameChangePatch.cs:40]: missing informal name assignment

File: source/GameInterface/Services/ObjectManager/ObjectManager.cs [hunk 6]
```diff
@@ -45,13 +44,21 @@ public interface IObjectManager
...
-    bool TryGetObject<T>(string id, out T obj) where T : class;
+    bool TryGetObject<T>(string id, out T obj) where T : MBObjectBase;
+    /// <summary>
+    /// Attempts to get an object not based on <see cref="MBObjectBase"/> object using a StringId and object type
+    /// </summary>
+    /// <typeparam name="T">Type of object</typeparam>
+    /// <param name="id">StringId used to lookup object</param>
+    /// <param name="obj">Out parameter for the object (null if not found)</param>
+    /// <returns>True if successful, false if failed</returns>
+    bool TryGetNonMBObject<T>(string id, out T obj) where T : class;
```
Comment (garrettluskey) [inline source/GameInterface/Services/ObjectManager/ObjectManager.cs]: Revert this change please. `bool TryGetObject<T>(string id, out T obj) where T : class;` should only exist

File: source/GameInterface/Services/Settlements/Patches/ChangeOwnerOfSettlementPatch.cs [hunk 7]
```diff
@@ -19,15 +19,10 @@ namespace GameInterface.Services.Settlements.Patches
 {
     /// <summary>
     /// Patches ownership changes of settlements
-    /// </summary>
+    /// </summary>ChangeOwnerOfSettlementPatch
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Patches/ChangeOwnerOfSettlementPatch.cs]:
```suggestion
    /// </summary>
```

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Coop.IntegrationTests/Environment/Mock/MockObjectManager.cs (hunk 1 above)
Comment (MaxDorob) [reply source/Coop.IntegrationTests/Environment/Mock/MockObjectManager.cs]: Reverted back remove reflection for IObjectManager

**Review by garrettluskey — APPROVED** [review]
